### ISSUE TITLE
Question: How can you map the screenshot key to another?

### ISSUE THREAD

**Issue by narukeh** [LukeSmithxyz/voidrice#1183]
Have a look at my keyboard layout:
![Lenovo_ThinkPad_L15_G2_AMD_7](https://user-images.githubusercontent.com/62762955/188308132-bb58fae9-f43b-4e85-afcb-955f23d07ee3.jpg)
Now i want to map the PrtSc to Super-right, and the second backspace that is above the calculator, top right corner, to the PrtSc to take screenshots.

Using xev, u can get keycodes or "nicknames", but only for the printscreen key it gives different output, here is the log:
https://termbin.com/fghg
https://pastebin.com/raw/NaPtZBvS
it gives one "headline" (one block of output) when i press Prtsc, and three other when i release the key.

**Comment by aymey** [comment]
can you not do this with `setxkbmap` and/or `xmodmap`?

**Comment by narukeh** [comment]
@aymey  sure, i dont care how, just that it works, the thing is, how are you going to refer to the screenshot key, even with xmodmap you would need a keycode/keysum,..........right? how would you suggest doing it?

**Comment by aymey** [comment]
Ill get home in about 6 hours to find the PrtSc keycode to use with `xmodmap` and hopefully give you an answer. But like i said erlier, `setxkbmap` doesnt need keycodes to work, you can reffer to keys by their name e.g. `setxkbmap -option caps:escape` to replace caps lock with the escape key. I cant check the PrtSc name though until i get home, sorry.

**Comment by aymey** [comment]
Okay, sorry i took so long but now that im home i can finally test my examples. You can use `xmodmap` to do this by
```
xmodmap -e "keysym Print = Super_R" -e "keysym Delete = Print"
```
The first expression signified with the `-e` flag (`"keysym Print = Super_R"`) maps `Print` (PrtSc) to `Super_R` (Right Super), the second expression (`"keysym Delete = Print"`) maps the `Delete` (Delete/Insert) key to the `Print` key.

Its important to note that your keysyms may be different, use `xmodmap -pke` to get a list or you can find the `keysym` section in `xev`. I hope this helps!

**Comment by narukeh** [comment]
>`Print`

ha! i tried guessing many names but they didnt work, even though it was simple.

Yes everything was clear, very well explained, i just didnt know to to refer to the `Print` key.

Thank you very much!!!
(cant test it right now, i will close this issue today though)

**Comment by narukeh** [comment]
@aymey  Didnt quite work. the print key is now super_r as desired.

I have two backspaces. the second one is above the calculator. they seem to have the same keycode/name "Backspace" / "keycode 22". 

When i try your command `xmodmap -e "keysym Print = Super_R" -e "keysym Delete = Print"` it makes the Delete key a screenshot, which i did not want. then i tried:
```bash
xmodmap -e "keysym Print = Super_R" -e "keysym Delete = Print"
```
but whis made **both** my *Backspace*s to screenshot keys.

Now half of what is desired, is accomplished. i gotta find a way to make **only** the second "Backspace" key to a screenshot key.

**Comment by LukeSmithxyz** [comment]
Oh my goodness, change the bind in dwm, not with xmodmap or some other weird command...

**Comment by narukeh** [comment]
It does not matter which way the problem is solved.
The thing is if i am to bind in dwm, the second backspace to run the maimpick script, even the main backspace wont work, because there is no way (that i know) to differentiate between these two same functional backspace-keys. (like mentioned in the screenshot, i got one main backspace key, and a second above the calculator/numpad. i like to mount the second one )

```bash
xev | tee /tmp/xev.log
```

outputs this: https://pastebin.com/U25pvCQE    ;     https://pastebin.pl/view/935cb768
important stuff from line 105 until 129, (i opened xev, presed once the main backspace key, then once the second backspace, then super+q to quit xev)

(i dont want to reopen the issue to be bothersome, but the issue was not understood and not solved)

i did map maimpick to super+f12, but then i have to select fullscreen. so i cant take a quick fulscreen screenshot with just one key (like its the default, cause i mapped the printscreen key to the super+L, because lenovo put it in its place, see screenshot)

### so to summarize i want my printscreen key to work as super+L, and my second backspace as my printscreen key.

@aymey  ✌️

**Comment by aymey** [comment]
before you start X, can you give the output of `showkey -s`. As mentioned in these sources ([1](https://askubuntu.com/questions/389353/how-can-i-set-a-certain-keycode-to-a-key-and-bind-it-to-the-right-action), [2](https://unix.stackexchange.com/questions/420528/mapping-key-using-xmodmap-changes-multiple-keys)). Unfortunately if both keys have the same scancode then, from what i could find, your only hope is to find a way to reset/modify the original keyboard mappings, if however  the keys have different keycodes, then you can use `setkeycodes`.

**Comment by narukeh** [comment]
`showkey -s` is pretty cool, but unfortunately they both have the same scancode. Guess ill figure sth out...
@aymey thank you very much for your involvement.
